src/librustc_resolve/late/diagnostics.rs

```diff
@@ -236,9 +236,11 @@ impl<'a> LateResolutionVisitor<'a, '_> {
         }
 
         // Try Levenshtein algorithm.
-        let levenshtein_worked = add_typo_suggestion(
-            &mut err, self.lookup_typo_candidate(path, ns, is_expected, span), ident_span
-        );
+        let typo_sugg = self.lookup_typo_candidate(path, ns, is_expected, span);
+        let def_span = typo_sugg.as_ref()
+            .and_then(|sugg| sugg.res.opt_def_id())
+            .and_then(|def_id| self.r.definitions.opt_span(def_id));
```

Comment:
BTW, I'm pretty sure that either implementation of `opt_span` or resolver can ask `cstore` for spans from other crates, so we can get these suggestions almost everywhere.

Reply:
@petrochenkov that was my original understanding, but I couldn't see any way of actually accomplishing that. We get the appropriate `DefId`, but `opt_span` certainly doesn't work. The only way that seemed to work was using `hir::Map`, which is of course not available in `resolve`.

Reply:
@estebank 
https://github.com/petrochenkov/rust/commit/cab814b406282a225c818d69f940a7075df320f1
(Needs cleanup.)